### PR TITLE
Feature/enhance minio storage

### PR DESCRIPTION
**Pull request by sijo5722-2021**
- Use minio object storage for loki index&chunks and longhorn backup 
- Remove aws s3 dependency for lonhorn backup 
- Configure disk volume size where docker volumes get stored
- Minio bucket, env specific users, access policy and lifecycle policy for loki and longhorn 
- Minio credentials stored in vault 
- Both loki and longhorn minio resources are created from control center post deploy , so that it exists even after the deletion of environments till the data get expires according to lifecycle policy set. It allows the logs to be referred , backups to be used even after env deletion.

### CODE REVIEW

File: terraform/config-params/control-center-post-config/variables.tf [hunk 1]
```diff
@@ -69,4 +69,14 @@ variable "netmaker_host_name" {
 
 variable "netmaker_version" {
   description = "netmaker version"
+}
+
+variable "loki_data_expiry" {
+  description = "number of days to expire minio loki bucket data"
```
Comment (dfry) [inline terraform/config-params/control-center-post-config/variables.tf:75]: do the bucket expiration settings need to be set here?  or can the application set them themselves?

**Review by dfry — COMMENTED** [review]
LGTM, please let me know if bucket retention time can be set by the application as opposed to at time of the bucket creation

File: terraform/config-params/control-center-post-config/variables.tf (hunk 1 above)
Comment (sijo5722-2021) [reply terraform/config-params/control-center-post-config/variables.tf:75]: Loki does not handle it , it delegates the purging of old chunks to the storage backend system. 
Lonhorn has provision to delete system backups , but it does not have a provision to run it in a scheduled manner, as per my initial observation.

Comment (muzammil360) [reply terraform/config-params/control-center-post-config/variables.tf:75]: @dfry @sijo5722-2021 is right about the expiry settings. Loki only manages the chunk objects lifecycle on the filesystem (using `compactor` and `table-manager`). The responsibility is delegated to the object store whenever it is being used.

**Comment by muzammil360** [comment]
> LGTM, please let me know if bucket retention time can be set by the application as opposed to at time of the bucket creation

https://grafana.com/docs/loki/latest/operations/storage/retention/
image

**Review by dfry — APPROVED** [review]
minor changes to make backwards compatible

**Pull request merged**